libsentrykube/config.py

```diff
@@ -67,14 +67,32 @@ class K8sConfig:
     materialized_manifests: str
 
     @classmethod
-    def from_conf(cls, conf: Mapping[str, Any]) -> K8sConfig:
+    def from_conf(cls, region_name: str, conf: Mapping[str, Any] | None) -> K8sConfig:
+        root = str(conf["root"]) if conf is not None and "root" in conf else "k8s"
+
+        cluster_def_root = (
+            str(conf["cluster_def_root"])
+            if conf is not None and "cluster_def_root" in conf
+            else f"clusters/{region_name}"
+        )
+
+        cluster_name = (
+            str(conf.get("cluster_name"))
+            if conf is not None and "cluster_name" in conf
+            else None
+        )
+
+        materialized_manifests = (
+            str(conf["materialized_manifests"])
+            if conf is not None and "materialized_manifests" in conf
+            else f"materialized_manifests/{region_name}"
+        )
+
```

Comment:
There are a 4 instances of checks of conf is not None here. Wonder if it would be better to do the conf is None check once and set the required fields in that scenario. Then the other scenarios don't need that check and the code would be easier to comprehend.

Reply:
Ya I was on the fence of the clarity of this vs setting it twice.